octobot_evaluators/evaluators/strategy_evaluator.py

```diff
@@ -220,19 +239,21 @@ def _get_available_node_paths(self,
 
     def _inner_get_available_node_paths(self, matrix_id, evaluator_type, exchange_name, cryptocurrency, symbol,
                                         use_cache=True):
-        return [
-            matrix.get_matrix_default_value_path(tentacle_name=evaluator,
-                                                 tentacle_type=evaluator_type,
-                                                 exchange_name=exchange_name,
-                                                 cryptocurrency=cryptocurrency,
-                                                 symbol=symbol,
-                                                 time_frame=time_frame)
-            for time_frame in self.get_available_time_frames(matrix_id, exchange_name, evaluator_type,
-                                                             cryptocurrency, symbol, use_cache=use_cache)
-            if common_enums.TimeFrames(time_frame) in self.strategy_time_frames
-            for evaluator in self._get_available_evaluators(matrix_id, exchange_name, evaluator_type,
-                                                            use_cache=use_cache)
-        ]
+        paths = []
+        for time_frame in self.get_available_time_frames(matrix_id, exchange_name, evaluator_type,
+                                                         cryptocurrency, symbol, use_cache=use_cache):
+            if common_enums.TimeFrames(time_frame) in self.strategy_time_frames:
+                for evaluator_name in self._get_available_evaluators(matrix_id, exchange_name, evaluator_type,
+                                                                     use_cache=use_cache):
+                    path = matrix.get_matrix_default_value_path(tentacle_name=evaluator_name,
+                                                                tentacle_type=evaluator_type,
+                                                                exchange_name=exchange_name,
+                                                                cryptocurrency=cryptocurrency,
+                                                                symbol=symbol,
+                                                                time_frame=time_frame)
+                    if matrix.get_tentacle_node(matrix_id, path) is not None:
+                        paths.append(path)
+        return paths
```

Comment:
```suggestion
        return [
        path := matrix.get_matrix_default_value_path(tentacle_name=evaluator,
                                                 tentacle_type=evaluator_type,
                                                 exchange_name=exchange_name,
                                                 cryptocurrency=cryptocurrency,
                                                 symbol=symbol,
                                                 time_frame=time_frame) if path is not None:
            for time_frame in self.get_available_time_frames(matrix_id, exchange_name, evaluator_type,
                                                             cryptocurrency, symbol, use_cache=use_cache)
            if common_enums.TimeFrames(time_frame) in self.strategy_time_frames
            for evaluator in self._get_available_evaluators(matrix_id, exchange_name, evaluator_type,
                                                            use_cache=use_cache)
        ]
```
I don't know if it's working

Reply:
XD not use about the `path := matrix.get_matrix_default_value_path`. I'll try it !

Reply:
Not working unfortunatelly, the := statement is not filtering (and we can't use it in comprehension "ifs" part:
![image](https://user-images.githubusercontent.com/9078616/159915217-20a8d6ca-ad6f-4249-8b73-7ec71ae06898.png)